a7/E_Vraj.c

```diff
@@ -0,0 +1,53 @@
+#include<stdio.h>
+#include<stdlib.h>
+#include<string.h>
+int parity_checker(int a, int b);
+
+int main(){
+    int N;
+    scanf("%d",&N);
+    int d[N];
+    int q=0;
+    int query[N];
+    int j=0;
+    int *a = malloc(sizeof(int)*1000000001);
```

Comment:
Malloc-ing an array of order 10<sup>9</sup> will consume 2-4 GB, which ~~definitely~~ probably exceeds the memory limit. Perhaps a more optimal solution exists without the need for such a large array.

Reply:
Ah sorry, that was code left over from a previous attempt. I just had to remove that line and it's for loop. The program doesn't rely on it at all.